src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  width: calc($cell-size * 7 + $gap * 6);
+  padding: 10px;
+
+  &__day {
+    box-sizing: border-box;
+
+    width: $cell-size;
+    height: $cell-size;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    text-align: center;
+
+    border: 1px black solid;
+
+    transition: 0.5s;
+    font-family: Arial, sans-serif;
+
+    background-color: #eee;
+  }
+
+  @for $day from 1 through 31 {
+    &__day:nth-child(#{$day})::before {
+      content: '#{$day}';
+      font-size: 30px;
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length-#{$length} > :nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+  
+  @each $name, $day in $days {
```

Comment:
The variable `$days` is used but not defined or imported in this file. Ensure that it is defined in a variables file or imported correctly.